### PR TITLE
Initial type declarations

### PR DESCRIPTION
**Pull request by dmvrtx**
- Introduce type definitions for some of the common components fo WCPay
 - Provide guidelines on adding new type definitions.
 
 ## Testing instructions
 
  - All JS tests should pass
  - Make sure to rebuild client with `npm run build:client` and take a quick tour of affected pages. Pages should load with content and do not introduce new JS errors.

### CODE REVIEW

**Review by kalessil — APPROVED** [review]
LGTM!

File: client/types/disputes.d.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	has_evidence: boolean;
+	due_by: number;
+}
+
+export interface Dispute {
+	status: string;
+	id: string;
+	evidence_details?: EvidenceDetails;
+	metadata: Record< string, string >;
+	productType: string;
```
Comment (KarlisJ) [inline client/types/disputes.d.ts:25]: This should not be part of the Dispute interface. 

If we look at https://github.com/Automattic/woocommerce-payments/blob/develop/client/disputes/evidence/index.js#L378 we'll see that productType for is taken from `dispute.metadata` object

```
let productType = dispute.metadata[ PRODUCT_TYPE_META_KEY ] || '';
```

Also, it's never used as property of dispute. 

This was taken from [a recent PR ](https://github.com/Automattic/woocommerce-payments/pull/2773/files#diff-340a2de153572380f61d1d4e3c43e10b6b07619748daa6dee1969fd9f1ee9d37R12) and I suspect that the mistake is already there. 

Checking against API response also affirms that `Dispute` doesn't have property `productType`.

Comment (kalessil) [reply client/types/disputes.d.ts]: Good catch, corrected in 0d2db988c861e1fbdb600b32267ee9d9facd18bb

File: client/types/disputes.d.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import { Charge } from './charges';
+
+interface Evidence {
+	[ key: string ]:
+		| string
+		| Record< string, boolean >
+		| Record< string, string >;
+	isUploading: Record< string, boolean >;
+	metadata: Record< string, string >;
+	uploadingErrors: Record< string, string >;
```
Comment (KarlisJ) [inline client/types/disputes.d.ts:12]: @kalessil 
API response doesn't contain these 3 keys. 

Looking at the code it seems that we later add them and pass them along to other components with additional fields. But honestly, I'm lacking experience with codebase to figure out from where and why are they coming from.

Comment (KarlisJ) [reply client/types/disputes.d.ts:13]: @dmvrtx I'm guessing these fields were taken from the same PR as others. Are we confident we actually need them? Unnecessary fields in definition could create a lot of confusion in the future.

Comment (dmvrtx) [reply client/types/disputes.d.ts:13]: Let's drop them from this type. If a need arises later, there is always a way to create a derivative type with those fields.

**Comment by kalessil** [comment]
I did manual tests, looks like the changes are not breaking anything.

**Comment by KarlisJ** [comment]
LGTM and ready to merge. 

Remove unused properties from Evidence and manual checks looks all good.

**Comment by kalessil** [comment]
:+1: , running E2E tests locally before merging.

**Pull request merged**